MobileBuy/buy/src/androidTest/java/com/shopify/buy/dataprovider/CompleteCheckoutTest.java

```diff
@@ -0,0 +1,204 @@
+package com.shopify.buy.dataprovider;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.shopify.buy.extensions.ShopifyAndroidTestCase;
+import com.shopify.buy.model.Checkout;
+import com.shopify.buy.model.PaymentToken;
+import com.shopify.buy.model.internal.CheckoutWrapper;
+
+import org.apache.http.HttpStatus;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.CountDownLatch;
+
+import okhttp3.HttpUrl;
+import okhttp3.Protocol;
+import retrofit2.Response;
+import rx.Observable;
+import rx.Subscriber;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.fail;
+
+/**
+ * The integration level test for the Mobile Buy SDK
+ */
+@RunWith(AndroidJUnit4.class)
+public class CompleteCheckoutTest extends ShopifyAndroidTestCase {
+
+    Checkout checkout;
+    CheckoutWrapper checkoutWrapper;
+
+    CheckoutRetrofitService checkoutRetrofitService;
+
+    PaymentToken paymentToken;
+
+    int callCount;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        checkout = Checkout.fromJson("{\"id\":100, \"channel\":\"mobile_app\",\"payment_due\":\"payment_due\",\"token\":\"checkout_token\"}");
+        checkoutWrapper = new CheckoutWrapper(checkout);
+
+        checkoutRetrofitService = Mockito.mock(CheckoutRetrofitService.class);
+
+        paymentToken = PaymentToken.createCreditCardPaymentToken("paymenttoken");
+
+        final Field retrofitServiceField = CheckoutServiceDefault.class.getDeclaredField("retrofitService");
+        retrofitServiceField.setAccessible(true);
+        retrofitServiceField.set((((BuyClientDefault) buyClient).checkoutService), checkoutRetrofitService);
+    }
+
+    @Test
+    public void testCompletingCheckoutWithNullCheckoutToken() throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
```

Comment:
agreed, removed them for all the parameter tests here and in the ShippingRatesTest